pegasus/sites.v3/code.org/views/visit_partial_event_logger.erb

```diff
@@ -0,0 +1,26 @@
+<%= view :analytics_event_log_helper, event_name: nil %>
+
+<script type='text/javascript'>
+  // Log visits to the following pages found in .md.partial files.
+  // Since parameters can't be passed in to an .erb view from a .md.partial file,
+  // this solution of associating urls with events was opted for to be able to
+  // specify which event to trigger based on the page the user is on when
+  // '{{ visit_partial_event_logger }}' is present.
+  const partialUrlToEventMap = {
+    '/ai': 'AI_PAGE_VISIT_EVENT',
+    '/educate/csc': 'CSC_CURRICULUM_PAGE_VISITED_EVENT',
+    '/maker': 'PHYSICAL_COMPUTING_PAGE_VISITED_EVENT'
```

Comment:
very creative solution! unfortunately, inline javascript is problematic in .erb files too. fortunately, I think you can do this without any js, by moving this map and logic into ruby code just above the `view :analytics_event_log_helper` call, and then passing the result in as the `event_name`. please give that a try and let me know if you run into any problems.

Reply:
Removed all javascript and just did ruby and it works! Thank you for the feedback!